tests/ProjectIdProviderTest.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+namespace Google\Auth\Tests;
+
+use Google\Auth\CredentialsLoader;
+use Google\Auth\ProjectIdProvider;
+use PHPUnit\Framework\TestCase;
+
+class ProjectIdProviderShellMock extends ProjectIdProvider
```

Comment:
Can you put test classes at the bottom of the file? You'll also need to wrap them in phpcs ignores:

```php
//@codingStandardsIgnoreStart
$stuff;
//@codingStandardsIgnoreEnd
```